### DISCUSSION TITLE
Disable TLS verification or provide CA cert with ureq

### DISCUSSION THREAD

**Discussion by Javex** [algesten/ureq#756] [Q&A]
Today a server I don't control started misbehaving: It is no longer sending its intermediate cert. To fix the issue quickly I wanted to either just disable TLS verification or explicitly provide the intermediate cert to the application. However, I can't figure out how to do either: I'm stuck on how to write a `TlsConnector`, because most of the implementation details that ureq uses are private, making it really difficult to come up with a way to make this simple change.

So for example, with `curl` I can do `curl -k`  (disable verification) or `curl --cacert ./cert.pem` to provide the intermediate cert and it just works. To emulate the latter with Rust I got as far as this:
```
fn get_tls_connector() -> Arc<dyn TlsConnector> {
    let cert = File::open("./cert.pem").unwrap();
    let mut cert = BufReader::new(cert);
    let cert = rustls_pemfile::read_one(&mut cert).unwrap().unwrap();
    let cert = match cert {
        rustls_pemfile::Item::X509Certificate(c) => c,
        _ => panic!("Not X.509 cert"),
    };
    let mut root_cert_store = RootCertStore::empty();
    root_cert_store.add(cert).unwrap();

    let client_config = rustls::ClientConfig::builder()
        .with_root_certificates(root_cert_store)
        .with_no_client_auth();

    Arc::new(Arc::new(client_config))
}
```

But this where I'm stuck. I've copied [this impl block](https://github.com/algesten/ureq/blob/main/src/rtls.rs#L88) and then replaced all error handling with `unwrap`, but because neither `TlsConnector` nor `ClientConfig` are part of my crate I can't write the `impl` block like this: `error[E0117]: only traits defined in the current crate can be implemented for types defined outside of the crate`

I _could_ go and fix this issue and also deal with the implementation of `ReadWrite` that I haven't done yet, but this is becoming an awful lot of code so I'm wondering if there's an easier way or if I'm doing something wrong? I'm fairly new to rust, so this might just be a newbie problem, or it could be that the ecosystem is young and some APIs haven't matured that far yet, so I'm hoping someone has an idea of what I should try next.

**Comment by algesten** [comment]
Hi! This sounds like an issue that been discussed before: https://github.com/algesten/ureq/issues/691
